mne/utils.py

```
@@ -790,6 +790,137 @@ def dec(*args, **kwargs):
+
+    Parameters
+    ----------
+    source : function
+        Function to copy the docstring from
+
+    Returns
+    -------
+    wrapper : function
+        The decorated function
+    """
+    def wrapper(func):
+        func.__doc__ = source.__doc__
+        return func
+    return wrapper
+
+
+def copy_function_doc_to_method_doc(source):
+    """Use the docstring from a function as docstring for a method.
+
+    This decorator is useful when implementing a method that just calls a
+    function. This pattern is prevalent in for example the plotting functions
+    of MNE. The first parameter specified in the docstring of the function is
+    removed in the docstring for the method (it is assumed to be set to
+    `self`):
+
+        >>> def plot_function(object, a, b):
+        ...     '''Docstring for plotting function.
+        ...     Parameters
+        ...     ----------
+        ...     object : instance of object
+        ...         The object to plot
+        ...     a : int
+        ...         Some parameter
+        ...     b : int
+        ...         Some parameter'''
+        ...     pass
+        ...
+        >>> class A:
+        ...     @copy_function_doc_to_method_doc(plot_function)
+        ...     def plot(self, a, b):
+        ...         plot_function(self, a, b)
+        >>> print(A.plot.__doc__)
+        Docstring for plotting function.
+            Parameters
+            ----------
+            a : int
+                Some parameter
+            b : int
+                Some parameter
+
+    Parameters
+    ----------
+    source : function
+        Function to copy the docstring from
+
+    Returns
+    -------
+    wrapper : function
+        The decorated method
+    """
+    def wrapper(func):
+        doc = source.__doc__.split('\n')
+
+        # Find parameter block
+        for line, text in enumerate(doc[:-2]):
+            if (text.strip() == 'Parameters' and
+                    doc[line + 1].strip() == '----------'):
```

Since the last commit, this will now fail loudly. If there is no parameter block, big error message.